src/Roslyn.Suppressions/Roslyn.Suppressions/InProcess/RoslynSettingsFileSynchronizer.cs

```diff
@@ -51,19 +51,19 @@ internal sealed class RoslynSettingsFileSynchronizer : IRoslynSettingsFileSynchr
         private readonly IThreadHandling threadHandling;
         private readonly IServerIssuesStore serverIssuesStore;
         private readonly IRoslynSettingsFileStorage roslynSettingsFileStorage;
-        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
```

Comment:
Swapped to use `IConfigurationProvider` rather than `IActiveSolutionBoundTracker` ("ASBT").

The ASBT uses the config provider internally and caches the value, which means it can be out of date when the binding state changes until the cached value is updated e.g. during migration.

The config provider re-calculates the value each time so it is always up to date i.e. as soon as the new format config files are written to disc the config provider will detect them.